synapse/api/auth.py

```diff
@@ -504,10 +512,9 @@ def _parse_and_validate_macaroon(
         try:
             macaroon = pymacaroons.Macaroon.deserialize(token)
         except Exception:  # deserialize can throw more-or-less anything
-            # doesn't look like a macaroon: treat it as an opaque token which
-            # must be in the database.
-            # TODO: it would be nice to get rid of this, but apparently some
-            # people use access tokens which aren't macaroons
+            # The access token doesn't look like a macaroon.
+            # In that case, we assume it's an opaque token which must be in the
+            # database.
```

Comment:
I think this gets called after we checked if it was in the database, so I'm not sure this is quite correct?

Reply:
I tried to write something and then wound up writing something which is easier to read in a different way. I'll try and rephrase.

Reply:
Maybe it's easier just to cut that sentence out.